### ISSUE TITLE
Abstract classes in ActionScript

### ISSUE THREAD

**Issue by joshtynjala** [apache/royale-compiler#73]
Proposed syntax:

``` actionscript
package com.example
{
	public abstract class MyAbstractClass
	{
		public MyAbstractClass()
		{
			
		}
	}
}
```

At compile time, use of an abstract class in a `new` statement should result in an error

``` actionscript
new MyAbstractClass(); //error
```

Similar to the `InterfaceCannotBeInstantiatedProblem` that is created when trying to instantiate an interface, there should be an problem for abstract classes with this message:

> Abstract classes cannot be instantiated with the new operator.

**Comment by joshtynjala** [comment]
Closed in commit 6a85bc1cd0c09e4ca07cf221a55ec9dd13e5d63c

**Comment by joshtynjala** [comment]
In bytecode, the class compiles with `[RoyaleAbstract]` metadata instead:

``` actionscript
package com.example
{
	[RoyaleAbstract]
	public class MyAbstractClass
	{
		public MyAbstractClass()
		{
			
		}
	}
}
```

The compiler is smart enough to look for this metadata to determine if a class from a compiled SWC file is abstract or not.
